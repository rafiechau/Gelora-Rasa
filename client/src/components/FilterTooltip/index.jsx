import React from 'react';
import PropTypes from 'prop-types';
import FilterListIcon from '@mui/icons-material/FilterList';
import { Tooltip, IconButton, Zoom } from '@mui/material';

const FilterTooltip = ({ onClick }) => (
  <Tooltip title="Filter" arrow TransitionComponent={Zoom}>
    <IconButton
      onClick={onClick}
      data-testid="filter-tooltip-button"
      sx={{
        color: 'white',
        backgroundColor: '#333',
        borderRadius: '1px',
        '&:hover': {
          backgroundColor: '#555',
          boxShadow: '0 2px 10px rgba(255, 255, 255, 0.3)',
        },
        '&:active': {
          backgroundColor: '#1a1a1a',
        },
      }}
    >
      <FilterListIcon />
    </IconButton>
  </Tooltip>
);

FilterTooltip.propTypes = {
  onClick: PropTypes.func.isRequired,
};

export default FilterTooltip;
