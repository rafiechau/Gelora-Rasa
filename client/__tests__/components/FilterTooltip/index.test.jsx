import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import FilterTooltip from '@components/FilterTooltip'; // Update the import path as needed

describe('FilterTooltip', () => {
  const mockOnClick = jest.fn();

  beforeEach(() => {
    render(<FilterTooltip onClick={mockOnClick} />);
  });

  test('should render the tooltip button', () => {
    expect(screen.getByTestId('filter-tooltip-button')).toBeInTheDocument();
  });

  test('should call onClick when the button is clicked', () => {
    fireEvent.click(screen.getByTestId('filter-tooltip-button'));
    expect(mockOnClick).toHaveBeenCalled();
  });

  test('should match the snapshot', () => {
    const tooltipButton = screen.getByTestId('filter-tooltip-button');
    expect(tooltipButton).toMatchSnapshot();
  });
});
