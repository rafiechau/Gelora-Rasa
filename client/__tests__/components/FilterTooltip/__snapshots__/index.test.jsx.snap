// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterTooltip should match the snapshot 1`] = `
<button
  aria-label="Filter"
  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-yyydyy-MuiButtonBase-root-MuiIconButton-root"
  data-mui-internal-clone-element="true"
  data-testid="filter-tooltip-button"
  tabindex="0"
  type="button"
>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
    data-testid="FilterListIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M10 18h4v-2h-4v2zM3 6v2h18V6H3zm3 7h12v-2H6v2z"
    />
  </svg>
  <span
    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
  />
</button>
`;
